trax/models/research/funnel_transformer.py

```diff
@@ -60,13 +62,23 @@ def SelectFirst():
   return tl.Fn('select_first', lambda x: x[:, 0, :])
 
 
-def _Upsample(short, long):
-  factor = -(-long.shape[1] // short.shape[1])  # ceil division
-  new_vecs = long + short.repeat(factor, axis=1)[:, :long.shape[1], :]
-  return new_vecs
+def _Upsampler(total_pool_size, separate_cls):
+  def _Upsample(short, long):
+    if separate_cls:
+      upsampled_short = jnp.append(
```

Comment:
maybe `jnp.concatenate` here? in vanilla numpy `append` calls `concatenate` under the hood if `axis` is given, who knows how it is implemented in `jnp`, so I guess it's better to express it directly that you want to concatenate

Reply:
done